charts/osm/crds/meshconfig.yaml

```diff
@@ -117,6 +117,34 @@ spec:
                       description: True for allowing traffic to flow between client and service pods within the mesh without SMI traffic policies, i.e. no traffic policy enforcement in the mesh. If set to false, enables deny-all traffic policy in mesh i.e. an SMI Traffic Target is necessary for services to communicate.
                       type: boolean
                       default: false
+                    inboundExternalAuthorization:
+                      description: Configures additional external authorization for inbound and ingress connections.
+                      type: object
+                      properties:
+                        enable:
+                          description: Enables/disables the inbound external authorization spec if present.
+                          type: boolean
+                          default: false
+                        address:
+                          description: Target destination endpoint that will handle external authorization.
+                          type: string
+                        port:
+                          description: Remote destination port for the external authorization endpoint.
+                          type: integer
+                          minimum: 1
+                          maximum: 65535
+                        statPrefix:
+                          description: String prefix for inbound external authorization related metrics.
+                          type: string
+                          default: "inboundExtAuthz"
+                        timeout:
+                          description: Defines the timeout to consider for the remote endpoint to reply in time.
+                          type: string
+                          default: "1s"
+                        failureModeAllow:
```

Comment:
All of these seem specific to what OPA needs. Might as well call the top level key something along the lines of `inboundExternalOPAAuth`

Reply:
The config for envoy proto is agnostic, and I think we should remain as such. Multiple external authorization services already exists, by just being in line with envoy we can support any of them (or at least subset, as long we can translate config)
https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto.html

Reply:
Thanks, that makes sense.